react/week3/react-w3/src/AddTodo.js

```diff
@@ -0,0 +1,46 @@
+import { useState } from "react";
+import Button from "./Button.js";
+function AddTodo({ todos, setTodos }) {
+  const [description, setDescription] = useState("");
+  const [deadline, setDeadline] = useState("");
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    if (!description) {
+      alert("Please provide a description");
+      return;
+    }
+    if (!deadline) {
+      alert("Please provide a deadline for your new todo");
+      return;
+    }
+    const newId = Math.max(...todos.map((item) => item.id)) + 1;
+    const obj = {
+      id: newId,
+      description: description,
+      deadline: deadline
+    };
+    setTodos([...todos, obj]);
+  }
```

Comment:
I think I would have defined a addTodo function where you also declared the todos (in App.js), and then passed it down here.
That way all of the functions that manipulate the todos are in one place.